Marlin/src/lcd/menu/menu.cpp

```diff
@@ -448,7 +450,7 @@ bool ui_selection; // = false
 void set_ui_selection(const bool sel) { ui_selection = sel; }
 void do_select_screen(PGM_P const yes, PGM_P const no, selectFunc_t yesFunc, selectFunc_t noFunc, PGM_P const pref, const char * const string/*=nullptr*/, PGM_P const suff/*=nullptr*/) {
   if (ui.encoderPosition) {
-    ui_selection = ((ENCODERBASE) > 0) == (int16_t(ui.encoderPosition) > 0);
+    ui_selection = ((encoderDirection > 0) == int16_t(ui.encoderPosition) > 0);
```

Comment:
As an experiment, try the commits I've added to see if it's better, and also add this here:
```cpp
  SERIAL_ECHOLNPAIR("ENCODERBASE=", int(ENCODERBASE),
                    " encoderDirection=", int(encoderDirection),
                    " encPos>0=`, int(ui.encoderPosition > 0)
  );
```
to see the values in action. If it's still broken, then attach your configurations to the reply so we can see why wrong values are coming out. From all these pieces we should be able to get it sorted quickly.

Reply:
My point is 
```
ui_selection = ((ENCODERBASE) > 0) == (int16_t(ui.encoderPosition) > 0);
```
This make the ui.selection REVERSED twice.

First  in ultralcd.h
```
 static constexpr int8_t encoderDirection = ENCODERBASE;
```

Second in the reverted commit.

All is good before. And yes, i dont know the intent of the reverted commit. But i will try.

Reply:
I forget who it was that needed the change, but it worked to fix the problem in their setup. Perhaps it needs refinement for keypads and other non-wheel setups too…?

Reply:
> This make the ui.selection REVERSED twice.

Presumably then, replacing `ENCODERBASE` with `encoderDirection` should do the trick. When in the selection screen, the direction should be as if inside an edit (or any non-menu) screen. I hope this tweak will do the trick.